framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/OpenShiftScenario.java

```diff
@@ -104,11 +116,18 @@ public final void deploy() {
     @Override
     public void undeploy() {
         try {
-            InstanceLogUtil.writeDeploymentLogs(this);
+            logger.info("Release log collector(s)");
+            try {
+                instancesLogCollectorFuture.cancel(false);
```

Comment:
The Executor should be shutdown, which will take care of individual tasks that have been submitted.

This way you take care of this individual task (there probably should not be more of them), but the documentation of this method suggests it might not always cancel the task. Thus I would recommend storing a reference to the ExecutorService and call:
`shutdown()`, `awaitTermination()` and `shutdownNow()` (if needed)

See https://stackoverflow.com/questions/10504172/how-to-shutdown-an-executorservice

Reply:
done

Reply:
I use shutdownNow as I don't need to wait for operations to be finished